components/calendar/src/astronomy.rs

```diff
@@ -0,0 +1,657 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// This file contains important structs and functions relating to location,
+/// time, and astronomy; these are intended for calender calculations and based off
+/// _Calendrical Calculations_ by Reingold & Dershowitz.
+use crate::iso::Iso;
+use crate::rata_die::RataDie;
+use crate::types::Moment;
+use crate::{Date, Gregorian};
+
+#[derive(Debug)]
```

Comment:
Anything that is public facing needs Rust doc strings to describe what they are (so `///` instead of `//` comments).